### DISCUSSION TITLE
Multiple instances/servers, refer entities from each other by UUID (autocomplete)

### DISCUSSION THREAD

**Discussion by slyk** [directus/directus#18114] [Archive]
_Hello. We got retail cosmetics shop with different tech used: opencart for online sell, drupal as central DB, custom client app for POS in offline shop, spreadsheets, sqlite-node for some additional microservices, etc._

Want to migrate from that tech zoo to Directus as DB model and Angular for client apps. But I guess it is a good idea to have **multiple instances of directus with different security and updgrade/change structure schedule**. 

I'm new to Directus, maybe someone can advice me more right way or have experience on similar projects?

So here is my plan:
- **core** server for rarely changed data structures and main entities (products, suppliers, retail shops configs). Max security, backups, strong restrictions to data write to that DB, only few persons can be access here;
- **work** server with more open write restrictions (list of orders from different shops, employee tasks list, vcards). With data being changed many times a day from different shops, but that can't affect others. Can be less secured, and less backups needed. If we lost some of orders/tasks we can live without it :)
- **microservices** server - used for additional business processes without strong connection to main DB. This is most often changed data structure, and least secured, because of outsource workers can be hired to work with different services. For example: notification services, shipping api adapters. If data from that server is lost it is ok, we can see that but nothing bad happens so we even will not need to restore backups.

So it looks ok to have multiple DBs on different VPSs/domains. But for UI it would be good to have some autocomplete. Each server will need to refer to products and other entities from core server (just read access). Microservices server need to refer all other servers entities (just by UUID). It's easy to make in own UI, but **maybe there is a way to make easy autocomplete field that refers to directus entites from other server?**

Sorry for my chaotic explanation of the question.

**Comment by rijkvanzanten** [comment]
There is! There's an autocomplete input field that you can use to query / select from any other third party (REST) API. However, do keep in mind that you won't get any of Directus' relational fetching/filtering etc, as that data doesn't "exist" within the current instance you're querying (it'd effectively just store the UUID, without being aware of what that relates to)

**Reply by slyk** [reply]
Thank you! I'll try this.
